### PR TITLE
Remove unused code and rearrange parameters to better reflect meaning

### PR DESCRIPTION
**Pull request by rpiaggio**
Namely, for inputs:

- `wavelength` is now `centralWavelength` and has been moved inside `mode` (only for spectroscopy, it's not required for imaging).
- `signalToNoiseAt` is now `atWavelength` and is mandatory instead of optional.

Code that only ran when `signalToNoiseAt` was absent has been removed.

### CODE REVIEW

File: modules/client/shared/src/main/scala/lucuma/itc/client/InstrumentMode.scala [hunk 1]
```diff
@@ -18,176 +19,169 @@ import lucuma.core.enums.GmosNorthGrating
...
         case (x0: GmosSouthImaging, y0: GmosSouthImaging)           => x0 === y0
         case _                                                      => false
-      }
 
-  final case class GmosNorthImaging(
-    filter: GmosNorthFilter
-  ) extends InstrumentMode
+  case class GmosNorthImaging(
+    filter:  GmosNorthFilter,
+    ccdMode: Option[GmosCcdMode]
```
Comment (cquiroz) [inline modules/client/shared/src/main/scala/lucuma/itc/client/InstrumentMode.scala:145]: Why did you add a param here??

Comment (rpiaggio) [reply modules/client/shared/src/main/scala/lucuma/itc/client/InstrumentMode.scala:145]: Because I noticed in the ITC servlet that Gmos Imaging can have CCD options, at least binning:

image

Comment (cquiroz) [reply modules/client/shared/src/main/scala/lucuma/itc/client/InstrumentMode.scala:145]: How will thiis affect the odb which calls imaging to calculate acq times?

File: modules/service/src/main/resources/graphql/itc.graphql [hunk 2]
```diff
@@ -10363,22 +10369,19 @@ input InstrumentModesInput {
 # Configuration alternatives query
 input SpectroscopyIntegrationTimeInput {
   # Observing wavelength.
```
Comment (cquiroz) [inline modules/service/src/main/resources/graphql/itc.graphql:10371]: Is the comment correct?

File: modules/client/shared/src/main/scala/lucuma/itc/client/InstrumentMode.scala (hunk 1 above)
Comment (rpiaggio) [reply modules/client/shared/src/main/scala/lucuma/itc/client/InstrumentMode.scala:145]: The parameter is optional. So, for acquisition, the ODB can just omit the parameter, which is equivalent to what was being done until now.  This just paves the way for when we invoke ITC for science imaging.

Unless binning parameters should be passed for computing acquisition? Do you know if this is the case? @swalker2m? @andrewwstephens?

File: modules/service/src/main/resources/graphql/itc.graphql (hunk 2 above)
Comment (rpiaggio) [reply modules/service/src/main/resources/graphql/itc.graphql:10371]: Probably not, it was already like that, but it's a good opportunity to improve it. I'll commit a fix.

**Review by cquiroz — APPROVED** [review]
LGTM
